packages/core/src/app/checkout/Checkout.tsx

```diff
@@ -253,6 +260,12 @@ class Checkout extends Component<
             }
 
             window.addEventListener('beforeunload', this.handleBeforeExit);
+
+            if (isCheckoutExtensionEnabled(data.getConfig()?.checkoutSettings)) {
+                setIsExtensionEnabled(true);
```

Comment:
I think you can directly read from the return value of `isCheckoutExtensionEnabled(data.getConfig()?.checkoutSettings)` and pass it as a prop via the `ExtensionContext` rather than having to set it to the state first and then read from it.

Reply:
As discussed offline, I have changed the experiment value reading method.
Therefore, the condition here is updated as:
https://github.com/bigcommerce/checkout-js/blob/994a53343bd3dc4af0f2407e7cf33076088c88f2/packages/core/src/app/checkout/Checkout.tsx#L264